Use path in moose-libmesh and update Linux compilers

milljm opened the pull request:
use path instead of allowing Conda to perform a git submodule update (it fails with relative paths??).
This change simply uses path, so we don't have to modify the submodule in any way. Thus preventing another hash being generated. This git commit hash never coincided with anything libMesh related. So it was difficult for developer to know precisely what version of libMesh they obtained via moose-libmesh conda package.

Also, this PR includes a bump to all the Linux compilers (from GCC 7.3.0 to 9.3.0).

Commit:
Use GCC 9.3.0. Use path instead of git_url

Strictly use the path set forth by our submodule instead of pulling from git.
Conda-build has an issue with relative paths in .gitmodules. It was necessary
to 'patch' this file before turning it over to conda-build. Thus, this would
generate a new hash, different from the one folks expect to see (and match)

Bump the linux compiler from 7.3.0 to 9.3.0 in Conda (for Linux)! This
requires all moose-packages (excluding moose-tools) to be 'bumped' as well.

Closes #16684 #16528

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/660100/) on 34cef85 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16702/site)

This comment will be updated on new commits.

cticenhour reviewed (changes requested):
Looks good to me, except for the merge commit. Could you remove that? Then I'd be happy to approve.

cticenhour reviewed (dismissed):
Thanks @milljm! Will merge when testing completes.

moosebuild commented:
Job [Create Linux Stack](https://civet.inl.gov/job/661024/) on 5be0b0e : invalidated by @cticenhour

Failed to download git source

cticenhour commented:
So what do you think is going on here @milljm? Both Linux and Mac stack creations are failing due to a step not being able to download via git. Do we just invalidate until we get lucky?

cticenhour commented:
Current error is: 

```
Warning: failed to download source.  If building, will try again after downloading recipe dependencies.
Error was: 
Command '['/usr/local/bin/git', 'clone', '--mirror', '/opt/civet/build_1/roystgnr/MetaPhysicL', '/opt/civet/build_1/conda_builds/conda_envs/pr-16702/conda-bld/git_cache/opt/civet/build_1/roystgnr/MetaPhysicL']' returned non-zero exit status 128.
Failed to download or patch source. Please see build log for info.
```

Commit:
remove git_url for path

Odd, the squash created a conflict of the obvious version line, but not the
git_url line. :shrug:, so this was missed during the first squash.

Closes #16684

cticenhour reviewed (approved):
LGTM!

cticenhour commented:
FYI - I want to wait until we get a merge through to devel before I merge this. There's a bit of a backup there right now.
